src/tm-threads.c

```diff
@@ -1291,7 +1334,17 @@ static int TmThreadKillThread(ThreadVars *tv)
     }
 
     /* join it and flag it as dead */
-    pthread_join(tv->t, NULL);
+#ifdef HAVE_DPDK
+    if (tv->lcore_id != LCORE_ID_ANY && tv->type == TVT_PPT) {
+        /* wait for DPDK worker threads */
+        rte_eal_wait_lcore(tv->lcore_id);
```

Comment:
there is no mandate to launch dpdk processing function with rte_ela_launch or rte_eal_remote_launch. Please rework